### ISSUE TITLE
Execution order of around + before/after hooks

### ISSUE THREAD

**Issue by UrsaDK** [icy-arctic-fox/spectator#12]
At the moment the order in which Spectator executes around + before/after hooks is:

```
before each
around each before
in the example
around each after
after each
```
[Spectator docs](https://gitlab.com/arctic-fox/spectator/-/wikis/Hooks#around_each):
> When mixed with the other "each" and "all" hooks, around_each starts execution after before_each and finishes execution before after_each.

This makes it difficult to do things like:

```
describe "example" do
  around_each do |proc|
    default_state = described_class.save_state
    proc.call
    described_class.load_state(default_state)
  end

  context "with state1" do
    before_each { described_class.load_state(state1) }
    it …
  end

  context "with state2" do
    before_each { described_class.load_state(state2) }
    it …
  end
end
```

Obviously there are work-arounds for the above, but I don't feel that they are as neat as using a combination of around + before/after hooks. 

The order of execution also differs from the order in which Rspec executes the same hooks.

[Rspec docs](https://relishapp.com/rspec/rspec-core/v/2-0/docs/hooks/around-hooks#before/after(:each)-hooks-are-wrapped-by-the-around-hook):
> around each before
> before each
> in the example
> after each
> around each after

I was wondering if there is a reason for this difference and if it would be possible to bring this bit of functionality into sync with Rspec?

**Comment by icy-arctic-fox** [comment]
TIL RSpec wraps before/after hooks with around.

There's no good reason for it to be the way it is. It works that way for convenience in the code.

I agree it should be changed to match RSpec. That is a breaking change, so it can't be in a patch release. It will be slated for v0.10.0. That release is still in the works. In the meantime, I'll try to get a branch you can use, if you want to avoid workarounds.

**Comment by UrsaDK** [comment]
No problems! I'm not pressed for time, so happy to wait for v0.10.0 to come out. Thanks for the offer though, much appreciated! :)

**Comment by icy-arctic-fox** [comment]
Implemented in v0.10
